js/index.js

```diff
@@ -257,25 +309,39 @@ const initializeBattle = () => {
 const battleActionHandler = (battleAction) => {
   let elem = document.querySelector('.battle-log')
   switch (battleAction) {
+
     case 'attack':
-      const damage = Math.floor((player.attack*10*Math.random())/(monster.defense*(0.8)));
+      let attackPower = player.attack+player.attackBonus;
+      let damage = Math.floor((attackPower*10*Math.random())/(monster.defense*(0.8)));
       monster.health = monster.health - damage;
       if (monster.health > 0) {
         console.log('You attack for', damage);
-        document.getElementById('monster-health-bar').style.width = monster.health + '%';
+        document.getElementById('monster-health-bar').style.width = Math.floor((monster.health/monster.maxHealth*100)) + '%';
         elem.innerHTML += 'You attack ' + monster.name + ' for ' + damage + ' health! </br>';
-        elem.scrollTop = elem.scrollHeight;
+        scrollLog(elem);
+        monsterAttack();
       } else {
         monster.health = 0;
         console.log(monster.name, 'is dead yo!');
         updateXPAndGoldAndEndBattle();
       }
       document.getElementById('monster-health').textContent = monster.health;
       break;
-    case 'hp-potion': console.log('You use a potion and restore 20 HP');
-      elem.innerHTML += 'You use a health potion, restoring 20 HP </br>' ;
-      elem.scrollTop = elem.scrollHeight;
+
+    case 'hp-potion':
+      console.log('You use a potion and restore 50 HP');
+      if (player.items.potions > 0 && player.health < player.maxHealth) {
+        player.health += 50;
+        player.items.potions = player.items.potions - 1;
+        document.getElementById('potion-text').innerHTML = player.items.potions;
+        if (player.health > player.maxHealth) player.health = player.maxHealth;
+        updateHealth();
+      }
+      elem.innerHTML += 'You use a health potion, restoring 50 HP </br>' ;
```

Comment:
This log line should only occur if you use the potion. same with line 332. ATM, if you don't have a potion, you don't heal, but the console still says you used one. It should be if(potion) -> do potion, else -> say you're out of potions. In that case, would you let the player reselect or just call the person a dumbass and let the monster attack? Should be handled carefully here